[VAULT-46135] Add docs for Terraform Secret Engine Root rotation (#2797)
hashicorp/web-unified-docs: content/vault/v2.x/content/api-docs/secret/terraform.mdx

  use. This token must have the needed permissions to manage all Organization,
  Team, and User tokens desired for this mount.

- `explicit_max_ttl` `(integer: 0)` – Specifies the maximum lifetime, in seconds,
  set on the root token in Terraform Cloud or Enterprise when Vault rotates it.
  Acts as an upper-bound safety net; Vault manages the root token lifecycle. The
  default (`0`) omits the expiration, so Terraform Cloud or Enterprise applies its
  default token lifetime of two years from the time of creation. The same value
  applies to both manual and automatic rotation. Refer to the Terraform token API
  documentation for
  [organization](/terraform/cloud-docs/api-docs/organization-tokens#request-body),
  [team](/terraform/cloud-docs/api-docs/team-tokens#request-body), and
  [user](/terraform/cloud-docs/api-docs/user-tokens#request-body) tokens for

>   details on the default expiration.

Reviewer: ```suggestion
- `explicit_max_ttl` `(integer: 0)` – When Vault manages and rotates the root
  token lifecycle for Terraform Cloud or Enterprise, `explicit_max_ttl` sets an
  explicit, maximum lifetime, in seconds, for root tokens. Use the default (`0`)
  to use the default Terraform token lifetime of two years from the time of
  creation. Vault uses `explicit_max_ttl` for manual and automatic token rotation.
  Refer to the Terraform token API documentation for
  [organization](/terraform/cloud-docs/api-docs/organization-tokens#request-body),
  [team](/terraform/cloud-docs/api-docs/team-tokens#request-body), and
  [user](/terraform/cloud-docs/api-docs/user-tokens#request-body) tokens for
  details on the default expiration.
```
The original text was a little hard for me to follow. Let me know if the edit is accurate or not and I'll adjust the suggestion as needed.

### Sample response

```json
{
  "data": {
    "address": "https://app.terraform.io",
    "base_path": "/api/v2/",
    "explicit_max_ttl": 0,
    "token_owner_type": "user",
    "owner_id": "user-glhf1234",
    "rotation_period": 0,
    "rotation_schedule": "0 0 * * SAT",
    "rotation_window": 3600,
    "disable_automated_rotation": false,
    "last_vault_rotation": "2026-04-04T00:00:00Z",
    "next_vault_rotation": "2026-04-11T00:00:00Z"
  }
}
```

The `token_owner_type` and `owner_id` fields describe the root token Vault
manages. Vault discovers them from Terraform Cloud or Enterprise the first time
it rotates the token, so they remain empty until the first rotation. Vault
includes the `last_vault_rotation` and `next_vault_rotation` fields only after a
rotation occurs or you configure a rotation schedule.

## Rotate root token

> This endpoint rotates the [configured root token](#configure-access) that Vault

Reviewer: ```suggestion
Rotate the [configured root token](#configure-access) that Vault
```
Style correction: avoid "this" as a pronoun

Typically we just state what the user can do with the endpoint (the job to be done) rather than introducing the endpoint as the thing that takes action

uses to authenticate with Terraform Cloud or Enterprise. Vault creates a new
token of the same type (organization, team, or user), validates it, stores it as
the new root token, and revokes the previous token where Terraform supports
revocation.

Use this endpoint for on-demand rotation. It does not require Vault Enterprise.
Configure scheduled, automatic rotation with the rotation fields on the
[`/terraform/config`](#configure-access) endpoint, which requires Vault

> Enterprise.

Reviewer: I don't understand what we're trying to say here. Does the endpoint handle on-demand rotation (CE and ENT) **and** configuring automatic rotation (ENT-only)?

To revoke the previous token during rotation, Vault must know its token ID. On
HCP Terraform and Terraform Enterprise v2.1 and later, Vault discovers the token
ID automatically from the `auth-token` link in the
[`/account/details`](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/account#sample-response)

> response, so it revokes the previous team or user token on every rotation.

Reviewer: ```suggestion
To revoke the previous instance of a token during rotation, Vault must know the
original token ID. With HCP Terraform and Terraform Enterprise v2.1+, Vault can
revoke the team or user token on every rotation because it can discover the token
ID automatically from the `auth-token` link in the
[`/account/details`](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/account#sample-response)
response.
```
Style correction: use short/simple sentences with directive language.

> On earlier versions of Terraform Enterprise that do not return the `auth-token`
> link, Vault cannot determine the ID of the initially-configured token. The first
> time Vault rotates a team or user token, it returns a warning asking you to revoke
> that initial token manually in Terraform. Vault records the ID of each token it
> creates, so subsequent rotations revoke the previous token automatically.

Reviewer: ```suggestion
If Vault cannot automatically determine the original token ID, it returns a
warning for you to revoke the original token in Terraform manually the first
time it rotates a team or user token. Vault records the ID of each token it
creates, so subsequent rotations revoke the previous token automatically.
```
We can simplify this paragraph

For organization tokens, Terraform implicitly replaces the token when Vault
creates a new one, so Vault performs no explicit revocation.

| Method | Path                     |
| :----- | :----------------------- |
| `POST` | `/terraform/rotate-root` |

### Parameters

> There are no parameters to this operation.

Reviewer: ```suggestion
None.
```
